dropwizard-core/src/main/java/io/dropwizard/core/cli/ServerCommand.java

```diff
@@ -50,6 +50,13 @@ protected void run(Environment environment, Namespace namespace, T configuration
             server.addEventListener(new LifeCycleListener());
             cleanupAsynchronously();
             server.start();
+            new Thread(() -> {
+                try {
+                    server.join();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }, "dw-awaiter").start();
```

Comment:
What about only doing this when virtual thread flag is enabled for both admin and app connectors? And maybe have a new function called something like `runBlocking` such that the user can have better control?

Reply:
@cty123 IMHO joining the server had to be done in all earlier versions, so I'd want to integrate this without an extra check.

The `runBlocking` method could be integrated. Do you have a special use case why someone would want to override this method?

Reply:
I think it would be helpful if the user is able to know when the server has shutdown, such that the user can perform some cleanups right after runBlocking function exits. Currently this is also achievable via shutdown hook, so it's not absolutely necessary.

Reply:
That's already possible with `ConfiguredCommand#cleanup()`. If that's the only concern, we shouldn't duplicate that method.